source/core/GenericFunctions.cpp

```diff
@@ -2437,6 +2437,84 @@ VIREO_FUNCTION_SIGNATURE4(ConvertEnum, StaticType, void, StaticType, void)
     return _NextInstruction();
 }
 //------------------------------------------------------------
+// arguments: output cluster, then variable number of inputs that can be null, error cluster or a 1D array of error clusters
+struct MergeErrorsParamBlock : public VarArgInstruction
+{
+    _ParamDef(ErrorCluster, ErrorClusterOut);
+    _ParamImmediateDef(StaticTypeAndData, ErrorClusterInputs[1]);
+    NEXT_INSTRUCTION_METHODV()
+};
+
+static Boolean UpdateErrorWhenErrorOrWarningFound(ErrorCluster* errorClusterIn, ErrorCluster* errorClusterOut, Boolean findErrors) {
+    Boolean errorFound = errorClusterIn->status && findErrors;
+    Boolean warningFound = !errorClusterIn->status && errorClusterIn->code != 0 && !findErrors;
+    if (errorFound || warningFound) {
+        errorClusterOut->status = errorClusterIn->status;
+        errorClusterOut->code = errorClusterIn->code;
+        errorClusterOut->source->AppendStringRef(errorClusterIn->source);
+        return true;
+    }
+    return false;
+}
+
+static Boolean ErrorOrWarningFound(StaticTypeAndData *inputs, IntIndex inputCount, ErrorCluster* errorClusterOut, Boolean findErrors) {
+    VIREO_ASSERT(inputs != null);
+    VIREO_ASSERT(errorClusterOut != null);
+    for (IntIndex i = 0; i < inputCount; i++) {
+        if (inputs[i]._pData == NULL) {  // input parameter was not wired, move to the next one
+            continue;
+        }
+        TypeRef parameterType = inputs[i]._paramType;
+        if (parameterType->IsArray()) {
+            IntIndex rank = parameterType->Rank();
+            if (rank != 1) {
+                ConstCStr errorMessage = "MergeErrors only supports ErrorCluster or 1D Array of ErrorCluster as inputs";
+                THREAD_EXEC()->LogEvent(EventLog::kHardDataError, errorMessage);
+                return THREAD_EXEC()->Stop();
+            }
+            TypedArrayCoreRef errorClusterArray = *(TypedArrayCoreRef*)inputs[i]._pData;
+            IntIndex arrayLength = errorClusterArray->Length();
+            for (IntIndex i = 0; i < arrayLength; i++) {
+                ErrorCluster *errorCluster = (ErrorCluster*)errorClusterArray->BeginAt(i);
+                if (UpdateErrorWhenErrorOrWarningFound(errorCluster, errorClusterOut, findErrors)) {
+                    return true;
+                }
+            }
+        } else {
+            ErrorCluster *errorCluster = (ErrorCluster*) inputs[i]._pData;
+            if (UpdateErrorWhenErrorOrWarningFound(errorCluster, errorClusterOut, findErrors)) {
+                return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+//------------------------------------------------------------
+// MergeErrors function for error clusters, supports multiple inputs
+VIREO_FUNCTION_SIGNATUREV(MergeErrors, MergeErrorsParamBlock)
+{
+    // Initialize output error cluster to not error
+    ErrorCluster* errorClusterOut = _ParamPointer(ErrorClusterOut);
+    errorClusterOut->SetError(false, 0, "", false);
```

Comment:
Looks like it works if the first input is not an array and does not work otherwise with .gvi on Desktop. -- which is definitely confusing. I will file David a CAR on this.